teachertool/src/components/AddCriteriaButton.tsx

```diff
@@ -0,0 +1,26 @@
+/// <reference path="../../../built/pxtblocks.d.ts"/>
+
+import { getSelectableCatalogCriteria } from "../state/helpers";
+import { Button } from "react-common/components/controls/Button";
+import { showModal } from "../transforms/showModal";
+import { AppStateContext } from "../state/appStateContext";
+import { useContext, useMemo } from "react";
+
+interface IProps {}
+
+export const AddCriteriaButton: React.FC<IProps> = ({}) => {
+    const { state: teacherTool } = useContext(AppStateContext);
+
+    const hasAvailableCriteria = useMemo<boolean>(
+        () => getSelectableCatalogCriteria(teacherTool).length > 0,
+        [teacherTool.catalog, teacherTool.rubric]
+    );
+    return <Button
+            className="primary"
+            label={lf("Add Criteria")}
+            onClick={() => showModal("catalog-display")}
+            title={lf("Add Criteria")}
+            leftIcon="fas fa-plus-circle"
+            disabled={!hasAvailableCriteria}
```

Comment:
Styling nit: I think the "disabled" version of the button should have the text and circle dimmed as well. Having the text and symbol be such a dark black makes it feel like I should still be able to click it.

Reply:
Another styling nit we can fix later: It looks like button lost its "inline" styling. The design of this button is for it to look like it's part of the page, not a primary button:
![image](https://github.com/microsoft/pxt/assets/12176807/db38d25b-62c9-49b0-87c5-c56b96d1a61b)


This is consistent with the styling of the "Add notes" buttons on the Result page. On this page there may be many such buttons, so it would be visual overload to make them all primary:

![image](https://github.com/microsoft/pxt/assets/12176807/c8e80e07-e4db-465d-a8b7-6127253df0ae)